statsmodels/stats/contingency_tables.py

```diff
@@ -1402,3 +1484,1238 @@ def cochrans_q(x, return_object=True):
+        NOT_AVAILABLE = "Not Available From Rao-Scott Method"
+        if rows_are_multiple_response and columns_are_multiple_response:
+            spmi_test = self._test_for_single_pairwise_mutual_independence_using_rao_scott_2
+            table_p_value = spmi_test(self.row_factors[0], self.column_factors[0])
+            result = self._build_MRCV_result(table_p_value, NOT_AVAILABLE, "Rao-Scott", "SPMI")
+            return result
+        elif columns_are_multiple_response:
+            table_p_value = mmi_test(self.row_factors[0], self.column_factors[0])
+            result = self._build_MRCV_result(table_p_value, NOT_AVAILABLE, "Rao-Scott", "MMI")
+            return result
+        elif rows_are_multiple_response:
+            table_p_value = mmi_test(self.column_factors[0], self.row_factors[0])
+            result = self._build_MRCV_result(table_p_value, NOT_AVAILABLE, "Rao-Scott", "MMI")
+            return result
+        else:
+            single_response_table = Table(self.table)
+            return single_response_table.test_nominal_association()
+
+    @classmethod
+    def _extract_and_validate_factors(cls, column_factors, row_factors):
+        """
+        Make sure that the factors that the user passed into the initial MultipleResponseContingencyTable
+        initializer are valid and that the combination of single and multiple
+        response variables is currently supported. Also, since we don't currently
+        support multiple factors on a single axis we enforce that only one
+        has been passed.
+
+        Also de-duplicate factor level names and pivot factors to wide
+        orientation (if necessary)
+
+        Parameters
+        ----------
+        row_factors : list of Factor instances
+            Factors to use on the rows
+        column_factors : list of Factor instances
+            Factors to use in the columns
+        Return
+        ------
+        (Factor, Factor)
+            A single factor instance to use on the columns and a
+            single factor instance to use on the rows.
+        """
+        try:
+            try:
+                if len(row_factors) > 1:
+                    msg = "we don't currently support tables with more than one factor on the rows"
+                    raise NotImplementedError(msg)
+                row_factor = row_factors[0]
+            except TypeError:
+                if isinstance(row_factors, Factor):
+                    row_factor = row_factors
+                else:
+                    msg = "row_factors must be either a list of Factors or a Factor instance"
+                    raise NotImplementedError(msg)
+            try:
+                if len(column_factors) > 2:
+                    msg = "we don't currently support tables with more than one factor on the columns"
+                    raise NotImplementedError(msg)
+                column_factor = column_factors[0]
+            except TypeError:
```

Comment:
Actually this illustrates a question I meant to ask in the initial PR message -- how do you think about the tradeoff between input validation and performance? The MRCV methods have a few pretty specific requirements for the data (e.g. the frame orientation, using only 0/1's, the number of factors passed). But every `if len(x) == 2` type validation check costs some bit of performance. 

The idea behind this weird `try` structure was to avoid doing an if statement to check whether column factor is a Factor instance vs a list (calling `len` on a Factor instance throws the type error). The idea being that people should mostly be passing in a list (since the param name is pluralized) so in the "default" case you save one validation step (it's my understanding that you don't pay a performance cost for entering a `try` block as long there's no exception).

I think I probably over-optimized here...using trys instead of ifs will create a very small performance improvement and it makes the code substantially less readable. But at the same time I didn't want to presume that that little bit of performance doesn't matter to the user.

I also thought about maybe adding something like `validate=True` as a parameter and letting users pass a param to explicitly disable validation checks if they want to squeeze performance. But that also requires complicating the code.

How do you think about the tradeoff?